File: app/src/main/java/com/adityasubathu/vaccinationreminder/databaseHandler.java
package com.adityasubathu.vaccinationreminder;

import android.annotation.SuppressLint;
import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

public class databaseHandler {

    private myDbHelper myHelper;

    databaseHandler(Context context, String username) {
        myHelper = new myDbHelper(context, username);
    }


    public long insertData(String name, String dob, String gender) {
        myHelper.createNewTable();
        SQLiteDatabase dbb = myHelper.getWritableDatabase();  //opening database
        ContentValues contentValues = new ContentValues();
        contentValues.put(myDbHelper.NAME, name);
        contentValues.put(myDbHelper.DATE_OF_BIRTH, dob);
        contentValues.put(myDbHelper.GENDER, gender);

        return dbb.insert(myDbHelper.TABLE_NAME, null, contentValues);

    }

    String getData() {

        myHelper.createNewTable();
        SQLiteDatabase db = myHelper.getReadableDatabase();
        String[] columns = {myDbHelper.UID, myDbHelper.NAME, myDbHelper.DATE_OF_BIRTH, myDbHelper.GENDER};

        @SuppressLint("Recycle")

        Cursor cursor = db.query(myDbHelper.TABLE_NAME, columns, null, null, null, null, null);
        StringBuilder buffer = new StringBuilder();

        while (cursor.moveToNext()) {

            int cid = cursor.getInt(cursor.getColumnIndex(myDbHelper.UID));
            String name = cursor.getString(cursor.getColumnIndex(myDbHelper.NAME));
            String dob = cursor.getString(cursor.getColumnIndex(myDbHelper.DATE_OF_BIRTH));
            String gender = cursor.getString(cursor.getColumnIndex(myDbHelper.GENDER));
            buffer.append(cid).append("  ").append(name).append("  ").append(dob).append("  ").append(gender).append("  ");

        }
        return buffer.toString();

    }

    void delete(String childName) {
        myHelper.createNewTable();
        SQLiteDatabase db = myHelper.getWritableDatabase();
        String[] whereArgs = {childName};

        db.delete(myDbHelper.TABLE_NAME, myDbHelper.NAME + " = ?", whereArgs);
    }


    static class myDbHelper extends SQLiteOpenHelper {

        private static final String DATABASE_NAME = "vaccinationReminder.db";  // Database Name
        private static String TABLE_NAME;                                      // Table Name
        private static int DATABASE_Version = 1;                               // Database Version
        private static final String UID = "_id";                               // Column I (Primary Key)
        private static final String NAME = "Name";                             // Column II
        private static final String DATE_OF_BIRTH = "DOB";                     // Column III
        private static final String GENDER = "Gender";                         // Column IV
        //private Context context;

        myDbHelper(Context context, String username) {
            super(context, DATABASE_NAME, null, DATABASE_Version);
            TABLE_NAME = username;
            //this.context = context;
        }

        @Override
        public void onCreate(SQLiteDatabase db) {
            try {
                createNewTable();
            } catch (Exception e) {
                e.printStackTrace();
            }
        }

        @Override
        public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
            DATABASE_Version = newVersion;
        }

        void createNewTable() {
            String command = "CREATE TABLE IF NOT EXISTS " + TABLE_NAME + " (" + UID + " INTEGER PRIMARY KEY AUTOINCREMENT, " + NAME + " VARCHAR(255) ," + DATE_OF_BIRTH + " VARCHAR(225)," + GENDER + " VARCHAR(255));";
            SQLiteDatabase db = getWritableDatabase();
            db.execSQL(command);
        }
    }
}

File: app/src/main/java/com/adityasubathu/vaccinationreminder/DatabaseHandler.kt
package com.adityasubathu.vaccinationreminder

import android.annotation.SuppressLint
import android.content.ContentValues
import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

class DatabaseHandler internal constructor(context: Context?, username: String) {
    private val myHelper: MyDbHelper
    fun insertData(name: String?, dob: String?, gender: String?): Long {
        myHelper.createNewTable()
        val dbb = myHelper.writableDatabase //opening database
        val contentValues = ContentValues()
        contentValues.put(MyDbHelper.NAME, name)
        contentValues.put(MyDbHelper.DATE_OF_BIRTH, dob)
        contentValues.put(MyDbHelper.GENDER, gender)
        return dbb.insert(MyDbHelper.TABLE_NAME, null, contentValues)
    }

    val data: String
        get() {
            myHelper.createNewTable()
            val db = myHelper.readableDatabase
            val columns = arrayOf(MyDbHelper.UID, MyDbHelper.NAME, MyDbHelper.DATE_OF_BIRTH, MyDbHelper.GENDER)
            @SuppressLint("Recycle") val cursor = db.query(MyDbHelper.TABLE_NAME, columns, null, null, null, null, null)
            val buffer = StringBuilder()
            while (cursor.moveToNext()) {
                val cid = cursor.getInt(cursor.getColumnIndex(MyDbHelper.UID))
                val name = cursor.getString(cursor.getColumnIndex(MyDbHelper.NAME))
                val dob = cursor.getString(cursor.getColumnIndex(MyDbHelper.DATE_OF_BIRTH))
                val gender = cursor.getString(cursor.getColumnIndex(MyDbHelper.GENDER))
                buffer.append(cid).append("  ").append(name).append("  ").append(dob).append("  ").append(gender).append("  ")
            }
            return buffer.toString()
        }

    fun delete(childName: String) {
        myHelper.createNewTable()
        val db = myHelper.writableDatabase
        val whereArgs = arrayOf(childName)
        db.delete(MyDbHelper.TABLE_NAME, MyDbHelper.NAME + " = ?", whereArgs)
    }

    internal class MyDbHelper
    (context: Context?, private val TBLE_NAME: String) : SQLiteOpenHelper(context, DATABASE_NAME, null, DATABASE_Version) {
        override fun onCreate(db: SQLiteDatabase) {
            TABLE_NAME = TBLE_NAME
            try {
                createNewTable()
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }

        override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
            DATABASE_Version = newVersion
        }

        fun createNewTable() {
            val command = "CREATE TABLE IF NOT EXISTS $TBLE_NAME ($UID INTEGER PRIMARY KEY AUTOINCREMENT, $NAME VARCHAR(255) ,$DATE_OF_BIRTH VARCHAR(225),$GENDER VARCHAR(255));"
            val db = writableDatabase
            db.execSQL(command)
        }

        companion object {
            lateinit var TABLE_NAME: String
            const val DATABASE_NAME = "vaccinationReminder.db" // Database Name
            var DATABASE_Version = 1 // Database Version
            const val UID = "_id" // Column I (Primary Key)
            const val NAME = "Name" // Column II
            const val DATE_OF_BIRTH = "DOB" // Column III
            const val GENDER = "Gender" // Column IV
        }

    }

    init {
        myHelper = MyDbHelper(context, username)
    }
}